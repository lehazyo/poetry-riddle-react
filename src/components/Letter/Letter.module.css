.wrapper {
		display: flex;
		align-items: center;
		justify-content: center;
		position: relative;
		font-size: 16px;
		width: 30px;
		height: 30px;
		background-color: #dd2;
}

.letter {
		pointer-events: none;
}

.full {
		background-color: #2d2;
}

.inStone,
.inRiddles {
		position: absolute;
		right: 0;
		left: 0;
		height: 50%;
}

.inStone {
		top: 0;
		background: #aaa;
}

.inRiddles {
		bottom: 0;
		background: #bbb;
}

@media (min-width: 1000px) {
		.inStone,
		.inRiddles {
				height: auto;
				width: 50%;
		}
		.inStone {
				right: auto;
		}
		.inRiddles {
				left: auto;
		}
}