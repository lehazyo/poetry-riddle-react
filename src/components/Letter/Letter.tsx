import React from "react";
import styles from "./Letter.module.css";
import {Tooltip} from "@mui/material";
import cn from "classnames";

type TProps = {
    /** Displayed letter */
    letter: string,
    /** Letter is present in stone */
    inStone: boolean,
    /** Letter is used in answers to riddles */
    inRiddles: boolean,
}

/** Shows if letter is present in stone or in answers to riddles */
export const Letter = ({ letter, inStone, inRiddles }: TProps) => {
    if (!inStone && !inRiddles) {
        return null;
    }

    let tooltipText: string = "";
    if (inStone && inRiddles) {
        tooltipText = `Буква ${letter} использована вами на камне и в загадках`;
    } else {
        if (inStone) {
            tooltipText = `Буква ${letter} использована вами на камне, но не использована в загадках`;
        }
        if (inRiddles) {
            tooltipText = `Буква ${letter} использована вами в загадках, но не использована на камне`;
        }
    }

    return (
        <Tooltip title={tooltipText}>
            <div className={cn(styles.wrapper, { [styles.full]: inStone && inRiddles })}>
                <span className={styles.letter}>{letter}</span>
            </div>
        </Tooltip>
    )
};
