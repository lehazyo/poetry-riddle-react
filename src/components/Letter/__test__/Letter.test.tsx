import renderer from 'react-test-renderer';
import { Letter } from "../Letter";
import { cleanup, render } from '@testing-library/react';
import { screen } from '@testing-library/dom'

afterEach(cleanup);

describe("Letter tests", () => {
    it("Component matches snapshot", () => {
        const component = renderer.create(
            <Letter letter="А" inRiddles inStone/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Component is not rendered if inStone=false and inRiddle=false", () => {
        render(<Letter letter="А" inRiddles={false} inStone={false} />);
        expect(screen.queryByLabelText(/Буква/)).toBeNull();
        expect(screen.queryByText(/^А$/)).toBeNull();
    });

    it("Letter case is not changed", () => {
        render(<Letter letter="А" inRiddles inStone />);
        expect(screen.queryByText(/^А$/)).not.toBeNull();
        expect(screen.queryByText(/^а$/)).toBeNull();
    });

    it("Letter passed in 'letter' prop is rendered", () => {
        render(<Letter letter="Ж" inRiddles inStone />);
        expect(screen.getByText("Ж")).toBeTruthy();
    });

    it("If only inRiddles prop is specified, tooltip text is correct", () => {
        render(<Letter letter="Б" inRiddles inStone={false} />);
        expect(screen.queryByLabelText(/^Буква Б использована вами в загадках, но не использована на камне$/)).toBeTruthy();
    });

    it("If only inStone prop is specified, tooltip text is correct", () => {
        render(<Letter letter="Г" inRiddles={false} inStone />);
        expect(screen.queryByLabelText(/^Буква Г использована вами на камне, но не использована в загадках$/)).toBeTruthy();
    });

    it("If both inStone and inRiddles are specified, tooltip text is correct", () => {
        render(<Letter letter="Д" inRiddles inStone />);
        expect(screen.queryByLabelText(/^Буква Д использована вами на камне и в загадках$/)).toBeTruthy();
    });
});