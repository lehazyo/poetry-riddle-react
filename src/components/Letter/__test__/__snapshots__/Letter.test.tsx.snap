// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter tests Component matches snapshot 1`] = `
<div
  aria-label="Буква А использована вами на камне и в загадках"
  aria-labelledby={null}
  className="wrapper full"
  data-mui-internal-clone-element={true}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <span
    className="letter"
  >
    А
  </span>
</div>
`;
